python/pyspark/sql/connect/dataframe.py

```diff
@@ -45,6 +45,7 @@
     UnresolvedRegex,
 )
 from pyspark.sql.connect.functions import col, lit
+from pyspark.sql import Observation
```

Comment:
At first, because users are used to using the Observation from pyspark.sql. I think we better use it too.
Second, this PR only use the name of Observation, not any action of it.  So, the use is safely here.

Reply:
You can't use it for type annotations here if it's not legal to construct the type. In addition you're using it to access the name IIRC.

Reply:
https://github.com/apache/spark/pull/39091#discussion_r1073397082